test/FunctionalTests/Microsoft.OData.Client.Tests/Metadata/ClientTypeUtiTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ClientTypeUtilTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Client.Metadata;
+using System;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests.Metadata
+{
+    /// <summary>
+    /// A class test to test whether the use of Key Attribute on a Type property can identify a type as an entity when the conventional key naming methods are not used.
+    /// </summary>
+    public class ClientTypeUtilTests
+    {
+#if !PORTABLELIB
+        [Fact]
+        public void IFTypeProperty_HasKeyAttribute_TypeIsEntity()
+        {
+            //Arrange
+            Type person = typeof(Person);
+            bool expectedResult = true;
+            //Act
+            bool actualResult = ClientTypeUtil.TypeOrElementTypeIsEntity(person);
+            //Assert
+            Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void IFTypeProperty_HasNoKeyAttribute_TypeIsNotEntity()
+        {
+            //Arrange
+            Type student = typeof(Student);
+            bool expectedResult = false;
+            //Act
+            bool actualResult = ClientTypeUtil.TypeOrElementTypeIsEntity(student);
+            //Assert
+            Assert.Equal(expectedResult, actualResult);

```

Comment:
```suggestion
            Assert.IsFalse(expectedResult, actualResult);
```
Could assert directly to expected false value...